micro_rpc_build/src/lib.rs

```diff
@@ -22,6 +22,12 @@ use anyhow::Context;
 use prost_build::{Method, Service};
 use std::path::Path;
 
+#[derive(Default)]
+pub struct CompileOptions {
+    /// If true, server methods will use `&self` instead of `&mut self`.
+    pub immutable_server_methods: bool,
```

Comment:
In principle this may be extended to support other receivers (e.g. Arc, Pin, see https://doc.rust-lang.org/reference/items/associated-items.html#methods ), so I suggest making it an enum, also so that it is more semantically accurate (even with just two options, better than true / false), something like

```rust
enum ReceiverType {
  RefSelf,
  RefMutSelf,
}
````

Reply:
Done.